### DISCUSSION TITLE
Refactor code to use async version of Vapor API

### DISCUSSION THREAD

**Discussion by Kyle-Ye** [SwiftPackageIndex/SwiftPackageIndex-Server#1790] [Ideas]
I think it is needed as the Vapor 5 (The next version of Vapor) will not provide Eventloop API and will only provide async version API.

If we can process and refactor now. The cost of subsequent migration to Vapor 5 is expected to be much lower.

The last question is - Do we accept code refactor of the new async version of Vapor API currently?

**Comment by finestructure** [comment]
The refactor over to async/await already well under way. The reconcile and ingest commands are a/a already and most of the new functionality we add is a/a.

The only piece that's currently missing in Vapor is a working async command: https://discord.com/channels/431917998102675485/519613337638797315/978249416135823370

Help in making the transition is welcome! One thing I'd advise is to make the pull requests as localised and small as possible to reduce the chance of merge conflicts.

**Reply by finestructure** [reply]
There is a branch [`async-console`](https://github.com/SwiftPackageIndex/SwiftPackageIndex-Server/tree/async-console) where I've tried to adopt the recent async console-kit changes but run into the issues I discussed with Tim in the Vapor Discord channel linked above.

**Reply by Kyle-Ye** [reply]
Got the context. Thanks for sharing the information.

**Reply by daveverwer** [reply]
> Help in making the transition is welcome! One thing I'd advise is to make the pull requests as localised and small as possible to reduce the chance of merge conflicts.

If there's something you're specifically interested in converting, it's always good to chat with us about it first, too. Either here (this thread is great!) or [on our Discord server](https://discord.gg/vQRb6KkYRw).
